File: python/src/ziyy/scanner/token.py
from enum import auto, Enum

class TokenKind(Enum):
    LeftParen = 0
    RightParen = auto()
    Comma = auto()
    OpenTag = auto()
    CloseTag = auto()
    Dot = auto()
    Slash = auto()
    BackSlash = auto()
    # Literals.
    Identifier = auto()
    Number = auto()
    Text = auto()
    # Builtin Variables.
    Black = auto()
    Red = auto()
    Green = auto()
    Yellow = auto()
    Blue = auto()
    Magenta = auto()
    Cyan = auto()
    White = auto()
    Rgb = auto()
    B = auto()
    C = auto()
    I = auto()
    S = auto()
    U = auto()
    X = auto()
    # Keywords.
    Eof = auto()
    Error = auto()

class Token:
    def __init__(
        self,
        kind: TokenKind,
        content: str,
        err_code: int,
        line: int
    ) -> None:
        self.kind = kind
        self.content = content
        self.err_code = err_code
        self.err_code = err_code
        self.line = line

File: java/src/main/java/ziyy/scanner/Token.java
package ziyy.scanner;

public class Token {
    public TokenKind kind;
    public String content;
    public int errCode;
    public int line;

    Token(
            TokenKind kind,
            String content,
            int errCode,
            int line) {
        this.kind = kind;
        this.content = content;
        this.errCode = errCode;
        this.line = line;
    }
}
